packages/utils/telemetry-utils/src/sampledTelemetryHelper.ts

```diff
@@ -98,16 +109,27 @@ export class SampledTelemetryHelper implements IDisposable {
 	 * If no such distinction needs to be made, do not provide a value.
 	 * @returns Whatever the passed-in code block returns.
 	 */
-	public measure<T>(codeToMeasure: () => T, bucket: string = ""): T {
+	public measure<T extends LoggerData<TCustomMetrics>>(
```

Comment:
This is a breaking change for this API. There was not previously any constraints on what kind of data `codeToMeasure` could return. Now we require it return an interface with a `measurements` property. I don't think we can make this change without updating a lot of code in our repo. If we need this special API that observes some subset of properties coming out of the method call, I would recommend either:
* Creating a separate method that does that and has the extra type constraints
* Update the typing to not require the `measurements` field, and instead just look for the specified keys under `T`.

Happy to chat about these in more detail if that would help.